sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/b2c/AADB2CConfiguration.java

```diff
@@ -0,0 +1,159 @@
+                                 .clientAuthenticationMethod(ClientAuthenticationMethod.POST)
+                                 .authorizationGrantType(AuthorizationGrantType.AUTHORIZATION_CODE)
+                                 .redirectUri(properties.getReplyUrl())
+                                 .scope(properties.getClientId(), "openid", "offline_access")
+                                 .authorizationUri(AADB2CURL.getAuthorizationUrl(properties.getBaseUri()))
+                                 .tokenUri(AADB2CURL.getTokenUrl(properties.getBaseUri(), client.getValue()))
+                                 .jwkSetUri(AADB2CURL.getJwkSetUrl(properties.getBaseUri(), client.getValue()))
+                                 .userNameAttributeName(properties.getUserNameAttributeName())
+                                 .build();
+    }
+
+    /**
+     * Create client credential registration.
+     *
+     * @param client each client properties
+     * @return ClientRegistration
+     */
+    private ClientRegistration b2cClientCredentialRegistration(Map.Entry<String, AuthorizationClientProperties> client) {
+        AuthorizationGrantType authGrantType = Optional.ofNullable(client.getValue().getAuthorizationGrantType())
+                                                                .map(AADAuthorizationGrantType::getValue)
+                                                                .map(AuthorizationGrantType::new)
+                                                                .orElse(null);
+        if (authGrantType == null || !authGrantType.equals(AuthorizationGrantType.CLIENT_CREDENTIALS)) {
+            LOGGER.warn("Found {} registration of non client-credentials authorization type.", client.getKey());
+        }
+        return ClientRegistration.withRegistrationId(client.getKey())
+                                 .clientName(client.getKey())
+                                 .clientId(properties.getClientId())
+                                 .clientSecret(properties.getClientSecret())
+                                 .clientAuthenticationMethod(ClientAuthenticationMethod.POST)
+                                 .authorizationGrantType(authGrantType)
+                                 .scope(client.getValue().getScopes())
+                                 .tokenUri(AADB2CURL.getAADTokenUrl(properties.getTenantId()))
+                                 .jwkSetUri(AADB2CURL.getAADJwkSetUrl(properties.getTenantId()))
+                                 .build();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    @Conditional(AADB2CCondition.class)
+    public OAuth2AuthorizedClientManager authorizeClientManager(ClientRegistrationRepository clients,
+                                                         OAuth2AuthorizedClientRepository authorizedClients) {
+        DefaultOAuth2AuthorizedClientManager manager =
+            new DefaultOAuth2AuthorizedClientManager(clients, authorizedClients);
+
+        // @formatter:off
```

Comment:
Do we need this `@formatter:off` here?

Reply:
I just think this is not a normal indent，let me remove it.